R/biasCorrection.R

```diff
@@ -1,705 +1,966 @@
+                                                                     pr.threshold = pr.threshold,
+                                                                     n.quantiles = n.quantiles,
+                                                                     extrapolation = extrapolation,
+                                                                     theta = theta)
+                              }
+                              mem[,,,ind[i,1],ind[i,2]] <- end
+                        }
+                  }
+                  return(mem)
+            }) #end loop for members
+            #                   }else{
+            #                         ############# members jointly
+            #                   }
+            run[,,,,] <- abind(lmem, along = 2)
+            return(run)
+      }) #end loop for runtimes
+      bc$Data <- unname(abind(lrun, along = 1))
+      attr(bc$Data, "dimensions") <- attr(sim$Data, "dimensions")
+      ##########################
+      bc <- redim(bc, runtime = TRUE, drop = TRUE)
+      message("[", Sys.time(), "] Done.")
+      ##########################
+#      if(method == "variance" & method == "loci" & method == "ptr"){
+#        I=rep(0,nrow[dayListObs],ncol[bc$Data])
+#        lyidx = dayListObs[,1] ==2 & dayListObs[,2] ==29
+#        I[!lyidx,]=bc$Data
+#        I[lyidx,]=NaN
+#        rm(bc$Data)
+#        bc$Data=I
+#      }
+      bc$Dates <- sim$Dates
+      bc$InitializationDates <- sim$InitializationDates
+      bc$Members <- sim$Members
+      return(bc)
+}
+#end
+
+
+#' @title Bias correction methods on 1D data
+#' @description Implementation of several standard bias correction methods
+#' @param o A vector (e.g. station data) containing the observed climate data for the training period
+#' @param p A vector containing the simulated climate by the model for the training period. 
+#' @param s A vector containing the simulated climate for the variable used in \code{p}, but considering the test period.
+#' @param method method applied. Current accepted values are \code{"eqm"}, \code{"delta"},
+#'  \code{"scaling"}, \code{"gqm"}, \code{"gpqm"}, \code{"variance"}, \code{"loci"} and \code{"ptr"}. 
+#' @param scaling.type Character indicating the type of the scaling method. Options are \code{"additive"} (default)
+#' or \code{"multiplicative"} (see details). This argument is ignored if \code{"scaling"} is not selected as the bias correction method.
+#' @param precip Logical indicating if o, p, s is precipitation data.
+#' @param pr.threshold The minimum value that is considered as a non-zero precipitation. Ignored for
+#'  \code{varcode} values different from \code{"pr"}. Default to 1 (assuming mm). 
+#' @param extrapolation Character indicating the extrapolation method to be applied to correct values in  
+#' \code{"s"} that are out of the range of \code{"p"}. Extrapolation is applied only to the \code{"eqm"} metho
```

Comment:
I suggest replacing `flipud(gcmr)` by `apply(gcmr, 2, rev)`, so we avoid the dependency on package `matlab`